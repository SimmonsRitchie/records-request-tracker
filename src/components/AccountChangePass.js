import React from 'react';

export class AccountChangePass extends React.Component {
    state = {
        newPass1: "",
        newPass2: "",
        msg: ""
    }
    handleChangePass = () => {
            if (this.state.newPass1 !== this.state.newPass2) {
                this.setState(() => ({
                    msg: "Password fields don't match. Retype new password"
                }))   
            } else {
                this.props.user.updatePassword(this.state.newPass1).then(() => {
                        console.log("Password changed")    
                        this.setState(() => ({
                            msg: "Password changed"
                        }))   
                }).catch((error) => {
                        console.log("Password unchanged")
                        this.setState(() => ({
                            msg: error.message
                        }))
            })
        }
    }

    handleOnChange = (e) => {
        this.setState({ [e.target.name]: e.target.value });
    };

    render() {
        return (
            <div className="account__container">
                <input
                    name="newPass1"
                    type="password"
                    className="text-input"
                    placeholder="Enter new password"
                    onChange={this.handleOnChange}
                    value={this.state.newPass1}
                />
                <input
                    name="newPass2"
                    type="password"
                    className="text-input"
                    placeholder="Confirm new password"
                    onChange={this.handleOnChange}
                    value={this.state.newPass2}
                />
                {this.state.msg && <p className="form__error">{this.state.msg}</p>}
                <button
                    className="button"
                    onClick={this.handleChangePass}
                >
                    Set new password
                </button>
            </div> 
        )
    }
}
