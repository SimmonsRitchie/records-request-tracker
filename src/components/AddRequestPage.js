import React from 'react';
import RequestForm from './RequestForm'

const AddRequestPage = () => (
    <div>
    <p>Add request</p>
    <RequestForm />
    </div>
)

export default AddRequestPage;