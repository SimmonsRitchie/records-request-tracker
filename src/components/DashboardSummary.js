import React from 'react';
import {Link} from 'react-router-dom'


const DashboardSummary = () => (
    <div className="page-header">
        <div className="content-container">
            <h1 className="page-header__title">
                You have filed XX requests
            </h1>
            <h2 className="page-header__title">
                Viewing X because of filters
            </h2>
            <div className="page-header__actions">
                <Link className="button" to="/create">Add request</Link>
            </div>
        </div>
    </div>
)

export default DashboardSummary;