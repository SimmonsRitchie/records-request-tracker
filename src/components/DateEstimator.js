import React from 'react';
import dateEstimator from '../selectors/dateEstimator'

const DateEstimator = ({status, estInterimResponseDate, estFinalResponseDate, estAppealDeadline, estFinalDetermDate}) => (
    <div>
        <p className="date-estimator__title">
        Estimated timeline:
        </p>
        <p className="date-estimator__text">
            {dateEstimator(status, {
                estInterimResponseDate,
                estFinalResponseDate,
                estAppealDeadline,
                estFinalDetermDate,
            })}
        </p>
    </div>
)

export default DateEstimator;