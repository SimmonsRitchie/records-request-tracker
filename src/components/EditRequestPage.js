import React from 'react';
import { connect } from 'react-redux'
import RequestForm from './RequestForm'
import PageHeader from './PageHeader'
import { startEditRequest, startRemoveRequest } from '../actions/requests'
import RemovalModal from './RemovalModal'


/* What's happening here:
1. We pass down two props to RequestForm.
2. Firstly, we pass down the request info as the 'request' props. We get this info from redux.
In mapStateToProps we get data for the request that matches the ID of the 
URL. Eg. /edit/123 will get request data for the request with ID '123' 
3. Then, like AddRequestPage, we pass down the editRequest action dispatcher
to RequestForm under the 'onSubmit' prop.
4. When a user hits the 'save' button in RequestForm, it triggers onSubmit.
5. In this case, because we're feeding the 'editRequest' action instead of the
'addRequest' action, it triggers the 'editRequest' action when a user clicks the 'save' button.

*/

export class EditRequestPage extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            modalIsOpen: false
        };
    };
    startEditRequest = (request) => {
        this.props.startEditRequest(this.props.request.id, request)
        this.props.history.push('/')
    };
    openModal= () => {
        this.setState({modalIsOpen: true});
    }
    closeModal= () => {
        this.setState({modalIsOpen: false});
    }
    onRemove = (id) => {
        this.setState({modalIsOpen: false});
        this.props.startRemoveRequest(this.props.request.id);
        this.props.history.push('/')
    };
    render() { 
        return (
        <div>
            <PageHeader 
                pageTitle={"View/Edit request"}
            />
            <div className="content-container">
                <RequestForm 
                    onSubmit={this.startEditRequest}
                    request={this.props.request}
                />
            <button
                className="button button--remove"
                onClick={this.openModal}
            >
                Remove request
            </button>
            </div>
            <RemovalModal
                modalIsOpen={this.state.modalIsOpen}
                onRemove={this.onRemove}
                closeModal={this.closeModal}
            />
        </div>
        )
    }
}


// Request form is populated with the expense of the ID that matches
// the ID in URL (eg. /edit/123)
const mapStateToProps = (state, props) => {
    return {
        request: state.requests.find((request) => request.id === props.match.params.id)
    };
};

const mapDispatchToProps = (dispatch) => ({
    startEditRequest: (id, request) => dispatch(startEditRequest(id, request)),
    startRemoveRequest: (id) => dispatch(startRemoveRequest(id))
})

export default connect(mapStateToProps, mapDispatchToProps)(EditRequestPage);