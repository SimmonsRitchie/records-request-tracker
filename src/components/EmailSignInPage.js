import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom'

import LoginHeader from './LoginHeader'
import { emailSignIn, clearErrors } from '../actions/auth'
import { history } from '../routers/AppRouter'

class EmailSignInPage extends React.Component {
    state = {
        signInEmail: "",
        signInPass: "",
    }

    handleOnChange = (e) => {
        this.setState({ [e.target.name]: e.target.value });
    };

    handleOnSubmit = (e) => {
        e.preventDefault(); //Stops page from refreshing
        if (this.state.signInEmail !== "" && this.state.signInPass !== "") {
            this.props.emailSignIn(this.state.signInEmail, this.state.signInPass)
        }
    }

    handleBackButton = () => {
        this.props.clearErrors()
        history.push('/')
    }

    render() {

        const isInvalid = this.state.signInEmail === '' ||
            this.state.signInPass === ''


        return (
            <div className="box-layout">
                <div className="box-layout__box">
                    <LoginHeader />
                    <div className="login__button-container">
                        <form
                            className="login__button-container"
                            onSubmit={this.handleOnSubmit}
                        >
                            <input
                                name="signInEmail"
                                className="text-input bottom-margin"
                                placeholder="Enter email"
                                onChange={this.handleOnChange}
                                value={this.state.signInEmail}
                            />
                            <input
                                name="signInPass"
                                type="password"
                                className="text-input bottom-margin"
                                placeholder="Enter password"
                                onChange={this.handleOnChange}
                                value={this.state.signInPass}
                            />
                            {this.props.errorMsg && <p className="form__error">Error: {this.props.errorMsg}</p>}
                            <button
                                type="submit"
                                className="button bottom-margin"
                                disabled={isInvalid}
                            >
                                Sign-in
                            </button>
                        </form>
                        <Link
                            className="button bottom-margin"
                            to='/forgot'
                        >
                            Forgot password?
                        </Link>

                        <button
                            className="button button--secondary"
                            onClick={this.handleBackButton}
                        >
                            Back
                        </button>    
                </div>
            </div>
        </div>    
        )
    }
};

const mapStateToProps = (state) => ({
    errorCode: state.auth.signInErrorCode,
    errorMsg: state.auth.signInErrorMsg
})

const mapDispatchToProps = (dispatch) => ({
    emailSignIn: (email, pass) => dispatch(emailSignIn(email, pass)),
    clearErrors: () => dispatch(clearErrors())
});

export default connect(mapStateToProps,mapDispatchToProps)(EmailSignInPage);