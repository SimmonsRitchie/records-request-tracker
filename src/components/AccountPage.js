import React from 'react';
import {firebase} from '../firebase/firebase'
import PageHeader from './PageHeader'
import {AccountChangePass} from './AccountChangePass'


class AccountPage extends React.Component {
    state = {
        changePass: false
    }
    handleChangePass = () => {
        this.setState(() => ({
            changePass: !this.state.changePass
        }))   
    }

    render() {
        const user = firebase.auth().currentUser

        return (
            <div>
                <PageHeader 
                    pageTitle={"Account settings"}
                />
                <div className="content-container">
                    <h3>Email address: </h3>
                    <p>{user.email}</p>
                    <h3>Display name:</h3>
                    <p>{user.displayName ? user.displayName : "No display name set"}</p>
                    <button
                        className="button"
                        onClick={this.handleChangePass}
                    >
                        Change password
                    </button>
                    {this.state.changePass && <AccountChangePass user={user} />}
                </div>
            </div> 
        )
    }
}


export default AccountPage
