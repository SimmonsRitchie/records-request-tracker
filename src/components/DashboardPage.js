import React from 'react';
import RequestsListFilters from './RequestsListFilters'
import RequestsList from './RequestsList'
import DashboardSummary from './DashboardSummary';


const DashboardPage = () => (
    <div>
        <DashboardSummary />
        <RequestsListFilters />
        <RequestsList />
    </div>
);

export default DashboardPage;