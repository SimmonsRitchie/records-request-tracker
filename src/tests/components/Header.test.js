// Shallow rendering is different from 'full dom rendering', doesn't
// render child components
import ReactShallowRenderer from 'react-test-renderer/shallow';
import React from 'react';
import { Header } from '../../components/Header'
import { shallow } from 'enzyme'

// Using beforeEach, we can save ourselves from re-writing spies
// and the wrapper.
let startLogout, wrapper;

beforeEach(() => {
    startLogout = jest.fn();
    wrapper = shallow(<Header 
        startLogout={startLogout}
        />)
})

test('Should render Header correctly', () => {
    expect(wrapper).toMatchSnapshot()
});

test('Should call start Logout on button click', () => {
    // Below, we find the button tag and trigger 'onClick'
    // by simulating a click.
    wrapper.find('button').simulate('click')
    // We then check to see whether startLogout was called
    expect(startLogout).toHaveBeenCalled()
});