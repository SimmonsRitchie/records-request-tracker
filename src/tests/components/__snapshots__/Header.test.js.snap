// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Header correctly 1`] = `
<header
  className="header"
>
  <div
    className="content-container"
  >
    <div
      className="header__content"
    >
      <Link
        className="header__title"
        replace={false}
        to="/dashboard"
      >
        <h1>
          Right-To-Know tracker
        </h1>
      </Link>
      <div
        className="header__nav"
      >
        <Link
          replace={false}
          to="/rtkguide"
        >
          <button
            className="button button--nobg"
          >
            RTK guide
          </button>
        </Link>
        <button
          className="button button--nobg"
          onClick={[Function]}
        >
          Logout
        </button>
      </div>
    </div>
  </div>
</header>
`;
